engine/runtime-compiler/src/main/scala/org/enso/compiler/pass/analyse/FramePointerAnalysis.scala

```diff
@@ -286,9 +318,19 @@ case object FramePointerAnalysis extends IRPass {
 
   private def updateMeta(
     ir: IR,
-    framePointer: FramePointer
+    newMeta: FrameAnalysisMeta
   ): Unit = {
-    ir.passData().update(this, new FramePointerMeta(framePointer))
+    ir.passData().get(this) match {
+      case None =>
+        ir.passData()
+          .update(this, newMeta)
+      case Some(meta) =>
+        val ex = new IllegalStateException(
+          "Old: " + meta + " new " + newMeta
```

Comment:
```suggestion
          "Unexpected FrameAnalysisMeta associated with IR " + ir + "\nOld: " + meta + " new " + newMeta
```
would provide a bit more info immediately on the source of the problem.